lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -2034,6 +2040,11 @@ static void replaceUsesOfBuiltinVar(Value *V, const APInt &AccumulatedOffset,
                     Builder.CreateCall(ReplacementFunc, {Builder.getInt32(I)})),
                 Builder.getInt32(I));
           }
+          // Insert a bitcast from the reconstructed vector to the load vector
+          // type in case they are different.
+          // %2 = insertelement <3 x i64> %0, i64 %1, i32 2
+          // bitcast <3 x i64> %2 to <6 x i32>
```

Comment:
```suggestion
          // Input:
          // %2 = insertelement <3 x i64> %0, i64 %1, i32 2
          // %3 = extractelement <6 x i32> %2, i32 0
          // Modified:
          // %2 = insertelement <3 x i64> %0, i64 %1, i32 2
          // %3 = bitcast <3 x i64> %2 to <6 x i32>
          // %4 = extractelement <6 x i32> %3, i32 0
          
```

Reply:
The suggested comment doesn't exactly match what we are seeing, I took this idea and updated the comment.